mat64/vector.go

```diff
@@ -357,6 +360,61 @@ func (v *Vector) MulVec(a Matrix, b *Vector) {
 	}
 }
 
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+//
+// Vector is little-endian encoded as follows:
+//   0 -  8  number of elements     (int64)
+//   8 - ..  vector's data elements (float64)
+func (v Vector) MarshalBinary() ([]byte, error) {
+	buf := bytes.NewBuffer(make([]byte, 0, sizeInt64+v.n*sizeFloat64))
+	err := binary.Write(buf, defaultEndian, int64(v.n))
+	if err != nil {
+		return nil, err
+	}
+
+	for _, val := range v.mat.Data {
```

Comment:
The Dense marshaling is only done on the visible data. Vector should do the same. If it becomes useful to serialise computational state for checkpointing (I guess it will), then as @btracey says, that can be done by gob, though even that it fraught; if you gob encode two views of the same backing matrix do you expect on deserialisation for the views to still be backed by the same data? You might want that and the behaviour might suggest it, but I don't think we will even bother trying to achieve that.